packages/testing/src/TestUtils.ts

```diff
@@ -297,54 +283,54 @@ export async function createLocalDemoFullDidFromKeypair(
  * @returns A FullDid instance that is not yet written to the blockchain.
  */
 export async function createLocalDemoFullDidFromLightDid(
-  lightDid: LightDidDetails
-): Promise<FullDidDetails> {
+  lightDid: DidDetails
+): Promise<DidDetails> {
   const { identifier } = lightDid
-  const authKey = lightDid.authenticationKey
-  const encKey = lightDid.encryptionKey
-
-  const keys: PublicKeys = {
-    [authKey.id]: authKey,
-  }
-  if (encKey) {
-    keys[encKey.id] = encKey
-  }
+  const [authKey] = lightDid.authentication
 
-  return new FullDidDetails({
+  return {
     uri: DidUtils.getKiltDidFromIdentifier(identifier, 'full'),
-    keyRelationships: {
-      authentication: new Set([authKey.id]),
-      keyAgreement: encKey ? new Set([encKey.id]) : new Set([]),
-      assertionMethod: new Set([authKey.id]),
-      capabilityDelegation: new Set([authKey.id]),
-    },
-    keys,
-    serviceEndpoints: {},
+    authentication: [authKey],
+    keyAgreement: lightDid.keyAgreement,
+    assertionMethod: [authKey],
+    capabilityDelegation: [authKey],
     identifier,
-  })
+  }
 }
 
 // It takes the auth key from the light DID and use it as attestation and delegation key as well.
 export async function createFullDidFromLightDid(
   payer: KeyringPair,
-  lightDidForId: LightDidDetails,
+  lightDidForId: DidDetails,
   sign: SignCallback
-): Promise<FullDidDetails> {
-  const api = await BlockchainApiConnection.getConnectionOrConnect()
-  return FullDidCreationBuilder.fromLightDidDetails(api, lightDidForId)
-    .setAttestationKey(lightDidForId.authenticationKey)
-    .setDelegationKey(lightDidForId.authenticationKey)
-    .buildAndSubmit(sign, payer.address, async (tx) => {
-      await Blockchain.signAndSubmitTx(tx, payer, {
-        resolveOn: Blockchain.IS_IN_BLOCK,
-      })
-    })
+): Promise<DidDetails> {
+  const [authenticationKey] = lightDidForId.authentication
+  const tx = await Chain.generateCreateTxFromCreationDetails(
+    {
+      identifier: DidUtils.getIdentifierByKey(authenticationKey),
+      authenticationKey,
```

Comment:
this is a recurrent pattern I see, plus afaik you can't even have a creation extrinsic where the identifier is not derived from the authentication key – why don't we pull this into the function ?

Reply:
You’re on spot, I also wanted to make the identifier prop optional

Reply:
Done in 7a9453ca